### PR TITLE
[#8266]Java: Use streams to improve code readability in teammates.common.datatransfer.questions.*

### LINKED ISSUE #8266: Java: Use streams to improve code readability in teammates.common.datatransfer.questions.*
As described in issue #7838, there are many methods in most classes under ./src/main/java/teammates/common/datatransfer/questions/ which also use loops. I would like to propose to simplify them to use Java 8 streams.

I read through the past comments on the issue and agree that it would indeed be beneficial for future developers in that they improve code readability. 

I would like to seek approval before starting work. @damithc @thenaesh @LiHaoTan

### PR DESCRIPTION
**Pull request by zishxn**
Fixes #8266

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java [hunk 1]
```diff
@@ -71,16 +75,16 @@ public boolean extractQuestionDetails(
             Map<String, String[]> requestParameters,
             FeedbackQuestionType questionType) {
         String numOfRubricChoicesString = HttpRequestHelper.getValueFromParamMap(requestParameters,
-                                                                Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_NUM_COLS);
+                Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_NUM_COLS);
         String numOfRubricSubQuestionsString = HttpRequestHelper.getValueFromParamMap(requestParameters,
-                                                                     Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_NUM_ROWS);
+                Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_NUM_ROWS);
```
Comment (madanalogy) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java:80]: Were the changes to these indentations absolutely necessary?

Comment (zishxn) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java:80]: > Were the changes to these indentations absolutely necessary?

Hi @madanalogy,
No. The indentation changes are not necessary. I just checked without these changes and there was no stylecheck violation. Please ignore these indentation changes.

**Comment by madanalogy** [comment]
@zishxn We just realised that this code might become redundant with the release of v7. We'll be putting this PR on hold for now. In the meantime, might I suggest you take a look at #10042 :)

**Comment by wkurniawan07** [comment]
Closing as the corresponding issue is closed.

**Pull request closed without merging**